[Debug] Build fix for Linux platforms

csaavedra opened the pull request:
#### 7a335876b24b43f5f0986933fc2b97e310b796bf
<pre>
[Debug] Build fix for Linux platforms
<a href="https://bugs.webkit.org/show_bug.cgi?id=268736">https://bugs.webkit.org/show_bug.cgi?id=268736</a>

Reviewed by Chris Dumez.

GlyphBufferStringOffset is defined as unsigned in
non-CG builds, so the assertion is redundant.
* Source/WebCore/platform/graphics/WidthIterator.cpp:
(WebCore::resetGlyphBuffer):

Canonical link: <a href="https://commits.webkit.org/274133@main">https://commits.webkit.org/274133@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/be6e65d3bcd0a30cb0312960de3a9f27ef61d067

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/37855 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/16763 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/40112 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/40398 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/33678 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/39185 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/19424 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/14050 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/32021 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/38425 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/14127 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/33173 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/12324 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/12284 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/33846 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/41669 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/34343 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/34331 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/38149 "Found 1 new test failure: animations/animation-iteration-event-destroy-renderer.html (failure)") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/12861 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/10495 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/36327 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/14443 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/8526 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/13298 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/13896 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/platform/graphics/WidthIterator.cpp:
```diff
@@ -187,7 +187,9 @@ bool WidthIterator::hasExtraSpacing() const
 static void resetGlyphBuffer(GlyphBuffer& glyphBuffer, GlyphBufferStringOffset index)
 {
+#if USE(CG)
     ASSERT(index >= 0);
```
Would this work?
```
ASSERT(!std::is_signed_v<GlyphBufferStringOffset> || index >= 0);
```

csaavedra replied:
Doesn't work here, probably because it's just static analysis. Regardless, I'm a bit worried about having a data type that has a different signedness depending on the architecture. Not sure if it's a good idea, other bounding bugs might go unnoticed. Any reason why this needs to be signed in Apple ports?

csaavedra commented on Source/WebCore/platform/graphics/WidthIterator.cpp:
```diff
@@ -187,7 +187,9 @@ bool WidthIterator::hasExtraSpacing() const
+#if USE(CG)
     ASSERT(index >= 0);
+#endif
     auto unsignedIndex = static_cast<unsigned>(index);
```
Also this cast wouldn't be needed if not using CG.

cdumez replied:
But doesn't hurt?

csaavedra replied:
Yeah, just reviewing my own mediocre patch :)

csaavedra committed:
[Debug] Build fix for Linux platforms
https://bugs.webkit.org/show_bug.cgi?id=268736

Reviewed by Chris Dumez.

GlyphBufferStringOffset is defined as unsigned in
non-CG builds, so the assertion is redundant.
* Source/WebCore/platform/graphics/WidthIterator.cpp:
(WebCore::resetGlyphBuffer):

Canonical link: https://commits.webkit.org/274133@main
